speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DatabaseSettingsMenu.cpp

```diff
@@ -0,0 +1,261 @@
+static void LoadDefaultSettings()
+{
+    sprintf(m_dbsettings.Username, GfuiEditboxGetString(s_scrHandle, m_usernameControl));
+    sprintf(m_dbsettings.Password, GfuiEditboxGetString(s_scrHandle, m_passwordControl));
+    sprintf(m_dbsettings.Address, GfuiEditboxGetString(s_scrHandle, m_addressControl));
+    sprintf(m_portString, GfuiEditboxGetString(s_scrHandle, m_portControl));
+    sprintf(m_dbsettings.Schema, GfuiEditboxGetString(s_scrHandle, m_schemaControl));
+    m_dbsettings.UseSSL = GfuiCheckboxIsChecked(s_scrHandle, m_useSSLControl);
+}
+
+/// @brief        Loads the settings from the config file into the internal variables
+/// @param p_param The configuration xml file handle
+static void LoadConfigSettings(void* p_param)
+{
+    // Set the max time setting from the xml file
+    sprintf(m_dbsettings.Username, GfParmGetStr(p_param, PRM_USERNAME, GFMNU_ATTR_TEXT, nullptr));
+    sprintf(m_dbsettings.Password, GfParmGetStr(p_param, PRM_PASSWORD, GFMNU_ATTR_TEXT, nullptr));
+    sprintf(m_dbsettings.Address, GfParmGetStr(p_param, PRM_ADDRESS, GFMNU_ATTR_TEXT, nullptr));
+    sprintf(m_portString, GfParmGetStr(p_param, PRM_PORT, GFMNU_ATTR_TEXT, nullptr));
+    sprintf(m_dbsettings.Schema, GfParmGetStr(p_param, PRM_SCHEMA, GFMNU_ATTR_TEXT, nullptr));
+    sprintf(m_dbsettings.CACertFileName, GfParmGetStr(p_param, PRM_CERT, "CA", nullptr));
+    sprintf(m_dbsettings.PublicCertFileName, GfParmGetStr(p_param, PRM_CERT, "public", nullptr));
+    sprintf(m_dbsettings.PrivateCertFileName, GfParmGetStr(p_param, PRM_CERT, "private", nullptr));
+
+    // Match the menu buttons with the initialized values / checking checkboxes and radiobuttons
+    SynchronizeControls();
+}
+
+/// @brief Loads the user menu settings from the local config file or the default values will be loaded
+static void OnActivate(void* /* dummy */)
+{
+    // Retrieves the saved user xml file, if it doesn't exist the default values will be loaded
+    std::string strPath("config/DatabaseSettingsMenu.xml");
+    char buf[512];
+    sprintf(buf, "%s%s", GfLocalDir(), strPath.c_str());
+    if (GfFileExists(buf))
+    {
+        void* param = GfParmReadFile(buf, GFPARM_RMODE_STD);
+        // Initialize settings with the retrieved xml file
+        LoadConfigSettings(param);
+        return;
+    }
+    LoadDefaultSettings();
+}
+
+/// @brief Returns to the main menu screen
+static void GoBack(void* /* dummy */)
+{
+    GfuiScreenActivate(DataSelectionMenuInit(s_scrHandle));
+}
+
+/// @brief Checks if a connection can be established with the database.
+static void CheckConnection(void* /* dummy */)
+{
+    m_dbsettings.UseSSL = false;
+    bool connectable = false;
+    try
+    {
+        GfuiLabelSetText(s_scrHandle, m_dbStatusControl, "Connecting...");
+        connectable = SMediator::GetInstance()->CheckConnection(m_dbsettings);
+    }
+    catch (std::exception& e)
+    {
+    }
```

Comment:
why use it if you are going to leave it empty. Put some error message in it. Look at how SD does it with GfLogError()

Reply:
Made an error message